client/client_update_test.go

```diff
@@ -842,7 +865,32 @@ func TestUpdateRootRemoteCorruptedNoLocalCache(t *testing.T) {
 // because the fact that the timestamp hasn't changed should mean that we don't
 // have to re-download the root.
 func TestUpdateRootRemoteCorruptedCanUseLocalCache(t *testing.T) {
-	for _, testData := range waysToMessUpServer {
+	if testing.Short() {
+		t.Skip("skipping test in short mode")
+	}
+
+	waysToMessUpServerRoot := waysToMessUpServer
+	for _, roleName := range data.BaseRoles {
+		waysToMessUpServerRoot = append(waysToMessUpServer,
+			swizzleExpectations{
+				desc: fmt.Sprintf("no %s keys", roleName),
+				expectErrs: []interface{}{
+					&certs.ErrValidationFail{}, signed.ErrRoleThreshold{}},
+				swizzle: func(s *testutils.MetadataSwizzler, role string) error {
+					return s.MutateRoot(func(r *data.Root) {
+						r.Roles[roleName].KeyIDs = []string{}
+					})
+				}},
+			swizzleExpectations{
+				desc:       fmt.Sprintf("no %s role", roleName),
+				expectErrs: []interface{}{tuf.ErrNotLoaded{}},
+				swizzle: func(s *testutils.MetadataSwizzler, role string) error {
+					return s.MutateRoot(func(r *data.Root) { delete(r.Roles, roleName) })
+				}},
```

Comment:
these are great additions to the test and the one above.  Since it's the same code specifying the additional swizzler expectations, could we move these additional expectations into a helper function?